internal/rollup/rollup_bundle.bzl

```diff
@@ -235,16 +268,81 @@ def run_sourcemapexplorer(ctx, js, map, output):
+
+  ctx.actions.expand_template(
+      output = output,
+      template =  ctx.file._system_config_tmpl,
+      substitutions = {
+          "TMPL_entry_points": str(entry_points),
+          "TMPL_main_entry_point": main_entry_point,
+      })
+
 def _rollup_bundle(ctx):
-  rollup_config = write_rollup_config(ctx)
-  run_rollup(ctx, collect_es6_sources(ctx), rollup_config, ctx.outputs.build_es6)
-  _run_tsc(ctx, ctx.outputs.build_es6, ctx.outputs.build_es5)
-  source_map = run_uglify(ctx, ctx.outputs.build_es5, ctx.outputs.build_es5_min)
-  run_uglify(ctx, ctx.outputs.build_es5, ctx.outputs.build_es5_min_debug, debug = True)
-  umd_rollup_config = write_rollup_config(ctx, filename = "_%s_umd.rollup.conf.js", output_format = "umd")
-  run_rollup(ctx, collect_es6_sources(ctx), umd_rollup_config, ctx.outputs.build_umd)
-  run_sourcemapexplorer(ctx, ctx.outputs.build_es5_min, source_map, ctx.outputs.explore_html)
-  files = [ctx.outputs.build_es5_min, source_map]
+  if ctx.attr.additional_entry_points:
+    # Generate code split bundles if additional entry points have been specified.
+    # See doc for additional_entry_points for more information.
+    # Note: ".cs" is needed on the output folders since ctx.label.name + ".es6" is already
+    # a folder that contains the re-rooted es6 sources
+    rollup_config = write_rollup_config(ctx, output_format="cjs", additional_entry_points=ctx.attr.additional_entry_points)
+    code_split_es6_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.es6")
+    _run_rollup(ctx, collect_es6_sources(ctx), rollup_config, code_split_es6_output_dir)
+    code_split_es5_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs")
+    _run_tsc_on_directory(ctx, code_split_es6_output_dir, code_split_es5_output_dir)
+    code_split_es5_min_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.min")
+    run_uglify(ctx, code_split_es5_output_dir, code_split_es5_min_output_dir, config_name="%s" % ctx.label.name, is_directory=True)
+    code_split_es5_min_debug_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.min_debug")
+    run_uglify(ctx, code_split_es5_output_dir, code_split_es5_min_debug_output_dir, debug=True, config_name="%s-debug" % ctx.label.name, is_directory=True)
+    # Generate the SystemJS boilerplate/entry point files
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs.es6", ctx.outputs.build_es6)
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs", ctx.outputs.build_es5)
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs.min", ctx.outputs.build_es5_min)
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs.min_debug", ctx.outputs.build_es5_min_debug)
+    # There is no UMD bundle when code-splitting but we still need to satisfy the output
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs", ctx.outputs.build_umd)
+    # There is no source map explorer output when code-splitting but we still need to satisfy the output
```

Comment:
why not? could we produce source map explorer outputs for each chunk? how would someone figure out why they have so many bytes? (even harder of course because it could be due to an eager dependency defeating your code splitting)
doesn't have to be in this PR though

Reply:
good point. we could make source map outputs for each chunk.

Reply:
will leave that for another PR

Reply:
:+1: